ibc-clients/ics07-tendermint/src/client_state/validation.rs

```diff
@@ -190,3 +200,52 @@ where
 
     Ok(Status::Active)
 }
+
+/// Check that the subject and substitute client states match as part of
+/// the client recovery validation step.
+///
+/// The subject and substitute client states match if all their respective
+/// client state parameters match except for frozen height, latest height,
+/// trusting period, and chain ID.
+pub fn check_substitute<V>(
+    subject_client_state: &ClientStateType,
+    _ctx: &V,
```

Comment:
Maybe we can remove this parameter? This seems a local function.

Reply:
Removing the `_ctx` parameter here then requires updating the `check_substitute` definition in `impl<V> ClientStateValidation for ClientState` to include a generic parameter like so:
```
fn check_substitute(&self, _ctx: &V, substitute_client_state: Any) -> Result<(), ClientError> {
    check_substitute::<V>(self.inner(), substitute_client_state)
}
```
I think that's a bit more confusing, so I'm leaning towards keeping the unused parameter in the function signature.

Reply:
Your suggestion above looks fine to me. In my view, `check_substitute` is just a local function - it is not exposed to users.

Reply:
Ok, I went ahead and incorporated the suggested change: [f1e9e69](https://github.com/cosmos/ibc-rs/pull/1127/commits/f1e9e69c0fcc9ac6f188fb33ba2a860db928ff2a)